server/plugin.go

```diff
@@ -410,26 +410,27 @@ func (p *Plugin) restart() {
 
 func (p *Plugin) generatePluginSecrets() error {
 	needSaveConfig := false
-	if p.configuration.WebhookSecret == "" {
+	cfg := p.getConfiguration()
```

Comment:
My concern with the `getConfiguration` here is, if you have a invalid configuration (for example, manually change in the config.json), you will get an empty configuration here (because the OnConfigurationChange, check that is valid), this would wipe out all the other settings, valid or not.

Reply:
@jespino, can you elaborate on how this makes things worse? If the configuration is indeed invalid and we throw it away `OnConfigurationChange`, would it be correct to treat it as empty as a result?

Reply:
Perhaps we should validate the config during `onActivate` ?